src/slang-nodes/index.d.ts

```diff
@@ -0,0 +1,523 @@
+import type { YulFunctionDefinition } from './YulFunctionDefinition';
+import type { YulIdentifier } from './YulIdentifier';
+import type { YulIfStatement } from './YulIfStatement';
+import type { YulLabel } from './YulLabel';
+import type { YulLeaveStatement } from './YulLeaveStatement';
+import type { YulLiteral } from './YulLiteral';
+import type { YulParameters } from './YulParameters';
+import type { YulParametersDeclaration } from './YulParametersDeclaration';
+import type { YulPath } from './YulPath';
+import type { YulPathComponent } from './YulPathComponent';
+import type { YulPaths } from './YulPaths';
+import type { YulReturnsDeclaration } from './YulReturnsDeclaration';
+import type { YulStackAssignmentOperator } from './YulStackAssignmentOperator';
+import type { YulStackAssignmentStatement } from './YulStackAssignmentStatement';
+import type { YulStatement } from './YulStatement';
+import type { YulStatements } from './YulStatements';
+import type { YulSwitchCase } from './YulSwitchCase';
+import type { YulSwitchCases } from './YulSwitchCases';
+import type { YulSwitchStatement } from './YulSwitchStatement';
+import type { YulValueCase } from './YulValueCase';
+import type { YulVariableAssignmentStatement } from './YulVariableAssignmentStatement';
+import type { YulVariableDeclarationStatement } from './YulVariableDeclarationStatement';
+import type { YulVariableDeclarationValue } from './YulVariableDeclarationValue';
+import type { YulVariableNames } from './YulVariableNames';
+
+export type BinaryOperation =
+  | AdditiveExpression
+  | MultiplicativeExpression
+  | ExponentiationExpression
+  | AssignmentExpression
+  | BitwiseAndExpression
+  | BitwiseOrExpression
+  | BitwiseXorExpression
+  | ComparisonExpression
+  | EqualityExpression
+  | AndExpression
+  | OrExpression
+  | ShiftExpression;
+
+export type FunctionLike =
+  | ConstructorDefinition
+  | FallbackFunctionDefinition
+  | FunctionDefinition
+  | FunctionType
+  | ModifierDefinition
+  | ReceiveFunctionDefinition
+  | UnnamedFunctionDefinition;
+
+export type HubNode =
+  | ArrayValues
+  | AssemblyFlags
+  | CallOptions
+  | CatchClauses
+  | ConstructorAttributes
+  | ContractMembers
+  | EnumMembers
+  | ErrorParameters
+  | EventParameters
+  | FallbackFunctionAttributes
+  | FunctionAttributes
+  | FunctionTypeAttributes
+  | HexStringLiterals
+  | IdentifierPath
+  | ImportDeconstructionSymbols
+  | InheritanceTypes
+  | InterfaceMembers
+  | LibraryMembers
+  | ModifierAttributes
+  | NamedArguments
+  | OverridePaths
+  | Parameters
+  | PositionalArguments
+  | ReceiveFunctionAttributes
+  | SourceUnitMembers
+  | Statements
+  | StateVariableAttributes
+  | StringLiterals
+  | StructMembers
+  | TupleDeconstructionElements
+  | TupleValues
+  | UnicodeStringLiterals
+  | UnnamedFunctionAttributes
+  | UsingDeconstructionSymbols
+  | VersionExpressionSet
+  | VersionExpressionSets
+  | VersionSpecifiers
+  | YulArguments
```

Comment:
Note to self: we should probably export these types automatically from our API for each enum/its variants.

Reply:
this is a list of object with `items` but the list of variants could also be helpful

Reply:
to avoid confusion, I renamed it from `HubNode` to `CollectionNode`

Reply:
I see. Would that still help in cases like `StatementVariant`?

Reply:
instead of listing them I just pushed a commit that automatically creates this type